
# HG changeset patch
# User ganguin <ganguin>
# Date 1654600180 0
# Node ID 2e739b8c44c72cfcd603a34217c7ca8d8a99c6e3
# Parent  4d909088d95c2addd0309c2d81d89dae0c4537ea
Bug 1772719 - fix wayland-only build with mesa X11 support r=stransky

this patch fixes building firefox with
--enable-default-toolkit=cairo-gtk3-wayland-only when system mesa is build
with X11 support.

Without this patch, it only builds with mesa built without X11 support.

Differential Revision: https://phabricator.services.mozilla.com/D148347

diff --git a/widget/gtk/nsClipboard.cpp b/widget/gtk/nsClipboard.cpp
--- a/widget/gtk/nsClipboard.cpp
+++ b/widget/gtk/nsClipboard.cpp
@@ -34,12 +34,15 @@
 #include "mozilla/SchedulerGroup.h"
 #include "mozilla/StaticPrefs_widget.h"
 #include "mozilla/TimeStamp.h"
+#include "GRefPtr.h"
 #include "WidgetUtilsGtk.h"
 
 #include "imgIContainer.h"
 
 #include <gtk/gtk.h>
-#include <gtk/gtkx.h>
+#if defined(MOZ_X11)
+#  include <gtk/gtkx.h>
+#endif
 
 #include "mozilla/Encoding.h"
 
